prow/getting_started_kind_minikube.md

```diff
@@ -0,0 +1,109 @@
+* Install configured application into your project (``Settings -> Developer settings -> GitHub Apps -> YourAppName -> Edit -> Install App``)
+* Install docker and kind or minikube. 
+
+## Filling Webhook URL
+
+The deployment will work on your localhost machine, but it needs to connect to the GitHub application configured in the previous section. There are several methods to do this with limited access to the public IP. The simplest would be to use dedicated services webhook services. A good example of such a service could be [smee.io](https://smee.io/). Such services should be used only for testing and development purposes. Please visit [smee.io/new](https://smee.io/new) to generate a new channel. Follow the instruction to install the tool (using ``npm``). Paste the generated link in the ``Webhook URL`` config field in your application configuration.
+
+The ``Webhook URL`` should be configured with a secret. To generate a secret please run in the console:
+```
+$ openssl rand -hex 20
+```
+Save the result locally, and paste it in the field ``Webhook secret (optional)`` during GitHub App configuration.
+
+## Deployment on kind or minikube
+
+### kind preparation
+Start a new kind cluster using:
+```
+kind create cluster
+```
+After a proper startup, you need IP address assigned to kind. To get IP assigned to kind you can execute the following command:
+```
+$ kubectl --context kind-kind  describe service kubernetes
+```
+IP is listed in `Endpoints:` section. If there is any port, ignore it. Finally, execute `smee` command. Taking `172.18.0.2` as an example IP run:
+```
+$ smee -u https://smee.io/YourWebhookHashHere --target 172.18.0.2:30003/hook
+```
+Apply prow CRDs using:
+```
+$ kubectl apply --context kind-kind --server-side=true -f https://raw.githubusercontent.com/kubernetes/test-infra/master/config/prow/cluster/prowjob-crd/prowjob_customresourcedefinition.yaml
+```
+### minikube preparation
+Alternatively, of you use minikube:
+```
+$ minikube start
+```
+Due to a deticated `minikube ip` command, you do not need to inspect IP address assigned to minikube. Just run the following command:
+```
+$ smee -u https://smee.io/YourWebhookHashHere --target $(minikube ip):30003/hook
+```
+Apply prow CRDs using:
+```
+$ kubectl apply --server-side=true -f https://raw.githubusercontent.com/kubernetes/test-infra/master/config/prow/cluster/prowjob-crd/prowjob_customresourcedefinition.yaml
+```
+
+### Update the sample manifest
+The sample manifest you need to use is [starter-s3-kind.yaml](config/prow/cluster/starter/starter-s3-kind.yaml).
+There are several fields that has to be configured before applying the manifest:
+
+* The github app cert by replacing the `<<insert-downloaded-cert-here>>` string
+* The github app id by replacing the `<<insert-the-app-id-here>>` string
+* The hmac token by replacing the `<< insert-hmac-token-here >>` string
```

Comment:
We should mention that this is the secret that you previously generated and saved to the gh app webhook